Source/WebCore/platform/graphics/cg/ImageBufferIOSurfaceBackend.cpp

```diff
@@ -239,6 +252,49 @@ void ImageBufferIOSurfaceBackend::ensureNativeImagesHaveCopiedBackingStore()
     flushContext();
 }
 
+void* ImageBufferIOSurfaceBackend::updateBitmapCacheForRead()
+{
+    if (!m_bitmapCacheLock.has_value()) {
+        flushContext();
+        m_bitmapCacheLock = IOSurface::Locker { *m_surface, IOSurface::Locker::AccessMode::ReadOnly };
+    }
+    return m_bitmapCacheLock->surfaceBaseAddress();
+}
+
+CGImageRef ImageBufferIOSurfaceBackend::updateBitmapCacheImage()
+{
+    if (m_bitmapCacheImage)
```

Comment:
The life cycle of `m_bitmapCacheImage` is not clear to me.

This function is called only from `ImageBufferIOSurfaceBackend::copyNativeImageForDrawing()` which is called from `ImageBuffer::draw()` via `ImageBuffer::copyNativeImageForDrawing()`. I`mageBuffer::draw()` calls `finalizeDrawIntoContext()` which calls `ImageBufferIOSurfaceBackend::releaseBitmapCache()`. So I think there is not any caching here.

Maybe I miss something here. But why do we have `m_bitmapCacheImage` as a member of this class instead of having it the return value of this function?

Reply:
>Maybe I miss something here. But why do we have m_bitmapCacheImage as a member of this class instead of having it the return value of this function?

It is kind of a good question. I think the context is in the descriptions we've been having few weeks in slack and meetings. So the idea is that there's a sequence of patches and this would be useful in the upcoming ones.

Maybe I'll take a redo the patch sequence to get progress with the reviews..